action.yml

```diff
@@ -58,9 +61,52 @@ runs:
               }
             ]
           }")
-          
-        host=$(echo $branch | jq --raw-output '.endpoints[0].host')
-        branch_id=$(echo $branch | jq --raw-output '.endpoints[0].branch_id')
+
+        if [[ $branch == *"already exists"* ]]; then
+
+        # Get the branch id by its name. We list all branches and filter by name
+        branch_id=$(curl --silent \
+            "https://console.neon.tech/api/v2/projects/${PROJECT_ID}/branches" \
+            --header "Accept: application/json" \
+            --header "Content-Type: application/json" \
+            --header "Authorization: Bearer ${API_KEY}" \
+            | jq -r .branches \
+            | jq -c '.[] | select(.name | contains("'${BRANCH_NAME}'")) .id' \
+            | jq -r \
+            ) \
```

Comment:
I'm not sure why we are looking for the branch by name.

Reply:
the create branch action takes a name as a parameter, right?

So if the name already exists we can't create the branch